qcodes/instrument_drivers/stanford_research/SIM928.py

```diff
@@ -0,0 +1,264 @@
+                               get_cmd=partial(self.get_idn, i))
+            self.add_parameter('voltage{}'.format(i), unit='V',
+                               label="Output voltage of module {}".format(i),
+                               vals=vals.Numbers(-20, 20),
+                               get_cmd=partial(self.get_voltage, i),
+                               set_cmd=partial(self.set_voltage, i))
+            self.add_parameter('voltage{}_step'.format(i), unit='V',
+                               label="The step size when changing the voltage "
+                                     "smoothly on module {}".format(i),
+                               parameter_class=ManualParameter,
+                               vals=vals.Numbers(0, 20), initial_value=0.005)
+        self.add_parameter('smooth_timestep', unit='s',
+                           label="The delay between sending the write commands"
+                                 "when changing the voltage smoothly",
+                           parameter_class=ManualParameter,
+                           vals=vals.Numbers(0,1), initial_value=0.05)
+
+        super().connect_message()
+
+    def get_idn(self, i=None):
+        """
+        i:
+            Slot of the module whose id is returned
+        Returns:
+            A dict containing vendor, model, serial, and firmware.
+        """
+        try:
+            idstr = ''  # in case self.ask fails
+            if i is None:
+                idstr = self.ask('*IDN?')
+            else:
+                idstr = self.ask_module(i, '*IDN?')
+            # form is supposed to be comma-separated, but we've seen
+            # other separators occasionally
+            for separator in ',;:':
+                # split into no more than 4 parts, so we don't lose info
+                idparts = [p.strip() for p in idstr.split(separator, 3)]
+                if len(idparts) > 1:
+                    break
+            # in case parts at the end are missing, fill in None
+            if len(idparts) < 4:
+                idparts += [None] * (4 - len(idparts))
+        except:
+            logging.warning('Error getting or interpreting *IDN?: ' +
+                            repr(idstr))
+            idparts = [None, None, None, None]
+
+        # some strings include the word 'model' at the front of model
+        if str(idparts[1]).lower().startswith('model'):
+            idparts[1] = str(idparts[1])[5:].strip()
+
+        return dict(zip(('vendor', 'model', 'serial', 'firmware'), idparts))
+
+    def find_modules(self):
+        """
+        Queries the SIM900 mainframe, in which slots there is a module present.
+        :return: a list of slot numbers where a module is present (starting
+                 from 1)
+        """
+        CTCR = self.ask('CTCR?')
+        CTCR = int(CTCR) >> 1
+        modules = []
+        for i in range(1,10):
+            if CTCR & 1 != 0:
+                modules.append(i)
```

Comment:
The code assumes that if there is a module in the slot, it must be a SIM928, while this is not always the case. Having an additional check here for a SIM928 solves this problem. For example:

```python
model = self.get_idn(i)['model']
if model == 'SIM928:
    modules.append(i)
```